src/libasr/codegen/asr_to_c.cpp

```diff
@@ -94,30 +126,40 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
                 throw Abort();
             }
         } else {
+            std::string dims;
             if (ASRUtils::is_integer(*v.m_type)) {
                 headers.insert("inttypes");
                 ASR::Integer_t *t = ASR::down_cast<ASR::Integer_t>(v.m_type);
-                std::string dims = convert_dims_c(t->n_dims, t->m_dims);
-                std::string type_name = "int" + std::to_string(t->m_kind * 8) + "_t";
+                dims = convert_dims_c(t->n_dims, t->m_dims);
+                std::string type_name;
+                if (t->m_kind == 1) {
+                    type_name = "int8_t";
+                } else if (t->m_kind == 2) {
+                    type_name = "int16_t";
+                } else if (t->m_kind == 4) {
+                    type_name = "int32_t";
+                } else if (t->m_kind == 8) {
+                    type_name = "int64_t";
+                }
```

Comment:
The earlier code was better. I removed this and replaced it with the one which is deleted. Please restore.